pkg/deploy/registry/rest.go

```diff
@@ -6,17 +6,20 @@ import (
 	"time"
 
 	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
 	kcoreclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
 	"k8s.io/kubernetes/pkg/fields"
 	"k8s.io/kubernetes/pkg/watch"
 
+	"github.com/golang/glog"
 	"github.com/openshift/origin/pkg/deploy/api"
 	deployutil "github.com/openshift/origin/pkg/deploy/util"
 )
 
 var (
 	// ErrUnknownDeploymentPhase is returned for WaitForRunningDeployment if an unknown phase is returned.
 	ErrUnknownDeploymentPhase = errors.New("unknown deployment phase")
+	ErrTooOldResourceVersion  = errors.New("too old resource version")
```

Comment:
It's strange that you recreate the error here and return it in place of the previous error which is essentially the same error?